### ISSUE TITLE
[Feat]: Log1p transformer 

### ISSUE THREAD

**Issue by TremaMiguel** [solegalli/feature_engine#274]
**Is your feature request related to a problem? Please describe.**

When you want to apply a logarithmic transformation and data contains a zero value, calculating `log(1+x)` is more convenient. 

**Describe the solution you'd like**

This could be another argument in [LogTransformer](https://github.com/solegalli/feature_engine/blob/master/feature_engine/transformation/log.py#L144) that checks for zeros and apply `log(1 + x)`

```
from feature_engine import transformation as vt

tf = vt.LogTransformer(variables = ['LotArea', 'GrLivArea'], check_zeros=True)
```
under the hood basically applies 
```
if check_zeros:
    log_1p = True if (X[self.variables_] == 0).any().any() else False
```

or it can be a separate transformer 

```
from feature_engine import transformation as vt

tf = vt.Log1PTransformer(variables = ['LotArea', 'GrLivArea'], check_zeros=True)
```

**Comment by solegalli** [comment]
What would this transformer do if `check_zeroes` is `True`? add 1 to the variable and then apply the log?

I am aware that this is done quite frequently. I just think it is dangerous to add 1 to a variable without really inspecting the variable nature that I am kind of reluctant to encourage its use by implementing it in a transformer.

Also why +1 and not +10?

**Comment by TremaMiguel** [comment]
I think it would be better for the user to decide which constant `C` to add and handle it as a separate transformer. For example, 

```
class LogCpTransformer(BaseNumericalTransformer):
    """
    The LogCpTransformer() applies the transformation log( x + C) where C is a positive constant to 
    the input variable. Adding a C constant helps apply log transformation to a variable having zero
    or negative values. 
    
    The LogTransformer() only works with numerical non-negative values. If the variable
    contains a zero or a negative value after adding a constant C, the transformer will return an error.
```

**Comment by GLevV** [comment]
In `LogCpTransformer` we can just set C=0 by default and also add option 'auto' which will do something like that `C = abs(min(x)) + 1`. This will allow us to use log transformation on negative values.
Also this one is bad 
>If the variable contains a zero or a negative value after adding a constant C, the transformer will return an error.
because log(0+1) = 0, so you will always return error. The reason behind log1p is to avoid infs and nans that will arise in log(0) case, not in avoiding zeros.

**Comment by solegalli** [comment]
> If the variable contains a zero or a negative value after adding a constant C, the transformer will return an error.

that line checks the variable **before** the transformation.

**Comment by TremaMiguel** [comment]
@solegalli  agree on developing this feature to open a PR ?

**Comment by solegalli** [comment]
go for it :)

**Comment by solegalli** [comment]
closed in #281
